noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/common.nr

```diff
@@ -36,15 +36,19 @@ pub fn validate_inputs(public_call: PublicCallData) {
 }
 
 pub fn validate_public_call_non_revert(public_call: PublicCallData) {
-    assert(public_call.call_stack_item.public_inputs.reverted == false, "Public call cannot be reverted");
+    assert(public_call.call_stack_item.public_inputs.reverted == 0, "Public call cannot be reverted");
 }
 
 pub fn initialize_reverted_flag(
     previous_kernel: PublicKernelData,
     public_call: PublicCallData,
     circuit_outputs: &mut PublicKernelCircuitPublicInputsBuilder
 ) {
-    circuit_outputs.reverted = previous_kernel.public_inputs.reverted | public_call.call_stack_item.public_inputs.reverted;
+    if previous_kernel.public_inputs.end_non_revertible.reverted != 0 {
```

Comment:
So, we always take the revert code of the previous kernel if non-zero. If zero we take the result of this iteration.

Sounds correct.